app/src/main/java/org/fossasia/openevent/app/core/session/create/CreateSessionPresenter.java

```diff
@@ -0,0 +1,48 @@
+package org.fossasia.openevent.app.core.session.create;
+
+import org.fossasia.openevent.app.common.ContextManager;
+import org.fossasia.openevent.app.common.mvp.presenter.AbstractBasePresenter;
+import org.fossasia.openevent.app.common.rx.Logger;
+import org.fossasia.openevent.app.data.session.Session;
+import org.fossasia.openevent.app.data.session.SessionRepository;
+
+import javax.inject.Inject;
+
+import timber.log.Timber;
+
+import static org.fossasia.openevent.app.common.rx.ViewTransformers.dispose;
+import static org.fossasia.openevent.app.common.rx.ViewTransformers.progressiveErroneous;
+
+public class CreateSessionPresenter extends AbstractBasePresenter<CreateSessionView> {
+
+    private final SessionRepository sessionRepository;
+    private final Session session = new Session();
+
+    @Inject
+    public CreateSessionPresenter(SessionRepository sessionRepository) {
+        this.sessionRepository = sessionRepository;
+    }
+
+    @Override
+    public void start() {
+        // Nothing to do
+    }
+
+    public Session getSession() {
+        return session;
+    }
+
+    public void createSession() {
+        session.setTrack(ContextManager.getSelectedTrack());
+        session.setEvent(ContextManager.getSelectedEvent());
```

Comment:
This can cause infinite recursion. You should create an empty event and track and set IDs in it accordingly